src/js/heuristicblocking.js

```diff
@@ -124,25 +124,25 @@ HeuristicBlocker.prototype = {
       self = this;
 
     // ignore first-party requests
-    if (!tab_origin || origin == tab_origin) {
+    if (!utils.isThirdPartyDomain(request_origin, tab_origin)) {
```

Comment:
Should restore the `!tab_origin` portion of the check since `tab_origin` may in fact be falsy and we don't want to keep going in that case.